### PR TITLE
add indicator to lab-card

### LINKED ISSUE #811: Add indicator on lab card that execution has outputs attached to it
#### Bug, feature request, or proposal:

This is a feature.

#### What is the expected behavior?

It would be really nice if we could see, just by looking at a lab card, whether an execution has generated outputs or not. Maybe we can show a little icon or something in the execution item. Or, if it's feasible from a UI/UX point of view, add a little carousel with the first three items or so attached to that execution. It would be really neat if one could click on such an item and would end up in the preview view of that output right away.

However, I'd be fine with just adding an icon for now so we can see immediately which execution has an at least one output.

#### What is the current behavior?

Right now, the lab card only shows the lab information, plus the first couple executions attached to that lab:

screen shot 2018-05-08 at 12 32 52

#### What is the use-case or motivation for changing an existing behavior?

This change would make it easier to quickly see which executions are "interesting".

### PR DESCRIPTION
**Pull request by Jefiozie**
This PR will resolve #811

### CODE REVIEW

File: client/src/app/shared/execution-card/execution-card.component.ts [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
+
+  constructor(public outputFilesService: OutputFilesService) {}
+
   @Input() execution;
 
   @Output() view = new EventEmitter<Execution>();
-
   ExecutionStatus = ExecutionStatus;
+  ngOnChanges(simple: SimpleChanges): void {
+    if (simple.hasOwnProperty('execution') && simple.execution.currentValue !== null) {
```
Comment (PascalPrecht) [inline client/src/app/shared/execution-card/execution-card.component.ts]: I don't think we need to use `hasOwnProperty()` here as we don't expect `execution` to be on the prototype anyways.

File: client/src/app/shared/execution-card/execution-card.component.html [hunk 2]
```diff
@@ -17,6 +15,7 @@
     </ng-container>
     {{ execution.name || '#' + execution.id }}
   </div>
+  <mat-icon *ngIf="hasOutput$ | async" style="float:right">playlist_add_check</mat-icon>
```
Comment (PascalPrecht) [inline client/src/app/shared/execution-card/execution-card.component.html]: This LGTM, we just need to come up with a different style here :)

Maybe we can use some flexbox-fu to align the icon accordingly.

Comment (Jefiozie) [reply client/src/app/shared/execution-card/execution-card.component.html]: Do you like the icon btw? Or do you prefer a other one?

File: client/src/app/shared/execution-card/execution-card.component.html [hunk 3]
```diff
@@ -5,17 +5,23 @@
...
+      <ng-container [ngSwitch]="execution.status">
+        <mat-icon *ngSwitchCase="ExecutionStatus.Finished">done</mat-icon>
+        <mat-progress-spinner *ngSwitchCase="ExecutionStatus.Executing" color="accent" mode="indeterminate">
+        </mat-progress-spinner>
+        <mat-icon *ngSwitchCase="ExecutionStatus.Stopped">info</mat-icon>
+        <mat-icon *ngSwitchCase="ExecutionStatus.Failed">block</mat-icon>
+      </ng-container>
+      {{ execution.name || '#' + execution.id }}
+    </div>
+    <div class="ml-execution-has-outputs">
```
[the remark this answers is not in the document]
Comment (Jefiozie) [reply client/src/app/shared/execution-card/execution-card.component.html]: See my new commit think this is what you want

**Comment by Jefiozie** [comment]
Preview of how I made it.

image

**Comment by PascalPrecht** [comment]
Great, I think this is okay for now. I personally would probably put the icon back in the bottom right corner (with spacing) and maybe even add a `matToolTip` that says something like "this execution has outputs" or maybe even make the icon link to the execution's output view.

But this can be done in a following PR. 

Last but not least, this needs to be squashed into a single commit and follow our commit message conventions as you know! :)

Thank you man!

**Comment by Jefiozie** [comment]
> Great, I think this is okay for now. I personally would probably put the icon back in the bottom right corner (with spacing) and maybe even add a matToolTip that says something like "this execution has outputs" or maybe even make the icon link to the execution's output view.

Agree!
This is better for the UX so I will add it.  💪

**Comment by Jefiozie** [comment]
Result:
image
image

**Comment by PascalPrecht** [comment]
Really don't wanna be picky here, but I think the commit message should be `feat(...): ....`

Not sure if the changelog generator will pick it up otherwise..

**Comment by PascalPrecht** [comment]
Oh, and don't forget to put the 

```
Closes: #811
```

back in there :)

**Comment by Jefiozie** [comment]
Oops..

**Comment by Jefiozie** [comment]
Oops..😃

**Comment by PascalPrecht** [comment]
Sweet! Waiting for travis

**Comment by Jefiozie** [comment]
Thanks for the quick response!

**Pull request merged**